Saule/Queries/Pagination/PagedResult.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace Saule.Queries.Pagination
+{
+    /// <summary>
+    /// Simple implementation of IPagedResult
+    /// </summary>
+    /// <typeparam name="T">Type of collection item</typeparam>
+    public class PagedResult<T> : IPagedResult
```

Comment:
Ah I see you added an implementation as well, do we really need the interface still in that case?

Reply:
yeah, so i also like more generic implementation like this instead of untyped interface, but it gives some problems:
1. to apply filtering\sorting\paging for IQueryable\IEnumerable we need to have actual data. And if endpoint returns `PagedResult<T>` then we need to get `PagedResult<T>.Items` that is typed property. In case of interface we can easily get it as untyped. If it's typed one, then we need to use reflection or expression trees, to get actual `Items` as we don't know type in the code and can't get them. But it's doable and if you want, i can drop interface and update code to use `PagedResult<T>` instead of interface

2. we need to understand original endpoint response that can be `PagedResult<T>` as we use it for paging and understanding `LastPage` in `PaginationQuery`. And for actual sorting\paging\filtering, So we need to unwrap actual `PagedResults<T>.Items` to apply operations on it. And we need to have both original endpoint result (for paging) and unwrapped result (for actual page\sort\filter operations)

Reply:
I see the issue now. As far as I remember we don't support those filtering/sorting on regular `IEnumerable`, only the generic version (but I could be wrong). So if it's just about serialization, I think it won't matter that much?

Since all `IEnumerable<T>` also implement `IEnumerable` we can maybe have an internal property that drops the type parameter?

Reply:
Refactored that and mentioned new details in the comment at bottom